File: leetcode-everyday/贪心算法/665-非递减数列/665-非递减数列.js
/**
 * @param {number[]} nums
 * @return {boolean}
 */
 var checkPossibility = function(nums) {
  if(nums.length === 1) return true;
  let valid = 1;
  for(let i = 0; i < nums.length-1; i++) {
      let x = nums[i];
      let y = nums[i+1];
      if(x > y) {
          -- valid;
          if(valid < 0) {
              return false;
          }
          if(i > 0 && y < nums[i-1]) {
              nums[i+1] = x;
          }
      }
  }  
  return true;
};

File: leetcode-everyday/贪心算法/665-非递减数列/665-非递减数列.cpp
class Solution {
public:
    bool checkPossibility(vector<int>& nums) {
        int valid = 1;
        for(int i = 0; i < nums.size()-1; i++) {
            int x = nums[i], y = nums[i+1];
            if(x > y) {
                -- valid;
                if(valid < 0) return false;
                if(i > 0 && y < nums[i-1]) {
                    nums[i+1] = x;
                }
            }
        }
        return true;
    }
};